src/controllers/product.controller.js

```diff
@@ -105,12 +106,44 @@ export const addProduct = async (req, res) => {
     });
     res.status(201).json(product);
   } catch (error) {
-    res.status(500).json( error );
+    res.status(500).json(error);
+  }
+};
+export const getAllForSeller = async (req, res) => {
+  try {
+    const authHeader = req.headers.authorization;
+    if (!authHeader) {
+      return res.status(401).json({ message: "Authorization header missing" });
+    }
+    const token = authHeader.split(" ")[1];
+    const decodedToken = jwt.verify(token, process.env.SECRET_KEY);
+    const sellerId = decodedToken.userId;
```

Comment:
this token verification is already created from middleware/verifyToken, it's always a good practice to use it as a middleware, there is no need to rewrite it